modules/databases/mssql_database/main.tf

```
@@ -18,9 +18,16 @@ locals {
     for group_key, group in try(var.settings.db_permissions, {}) : group_key => {
       db_roles = group.db_roles
       db_usernames = flatten([
-        for lz_key, value in group.managed_identities : [
-          for managed_identity_key in value.managed_identity_keys : [var.managed_identities[lz_key][managed_identity_key].name]
-        ]
+        [for mi_key, mi_value in group : [
+          for value in mi_value.keys : [
```

```suggestion
          for value in try(mi_value.keys, managed_identity_keys, []) : [
```
I would keep the original structure to prevent a breaking change, Can you confirm this is working?

---

I can add it, but it will work with only the current landing zone, without lz_key because variables' structure is different. If you are okay with it - I will add it.

---

Hi @LaurentLesle,
I added the parameter as you asked, and fixed some conditions